tools/TestFx/DelegatingHandlers/HttpMockServer.cs

```diff
@@ -45,32 +45,30 @@ private HttpMockServer()
         {
         }
 
-        public static void Initialize(Type callerIdentity, string testIdentity)
-        {
-            Initialize(callerIdentity, testIdentity, GetCurrentMode());
-        }
-
         public static void Initialize(string callerIdentity, string testIdentity)
-        {
-            Initialize(callerIdentity, testIdentity, GetCurrentMode());
-        }
-
-        public static void Initialize(Type callerIdentity, string testIdentity, HttpRecorderMode mode)
-        {
-            Initialize(callerIdentity.Name, testIdentity, mode);
-        }
-
-        public static void Initialize(string callerIdentity, string testIdentity, HttpRecorderMode mode)
         {
             _servers = new List<HttpMockServer>();
             _assetNames = new AssetNames();
             _records = new Records(Matcher);
 
             CallerIdentity = callerIdentity;
             TestIdentity = testIdentity;
-            Mode = mode;
             Variables = new Dictionary<string, string>();
 
+            string testMode = FileSystemUtilsObject?.GetEnvironmentVariable(ConnectionStringKeys.AZURE_TEST_MODE_ENVKEY);
+            if (string.IsNullOrEmpty(testMode))
+            {
+                Mode = HttpRecorderMode.Playback;
+            }
+            else if (Enum.TryParse(testMode, true, out HttpRecorderMode recorderMode) && Enum.IsDefined(typeof(HttpRecorderMode), recorderMode))
+            {
+                Mode = recorderMode;
+            }
+            else
```

Comment:
Why should we have 3 situations here? Just if not null and the value is Record -> Record, else PlayBack

Reply:
The null condition can be merged into the last one. If the value is valid, then the value will be considered as the mode. Otherwise, Playback.